### PR TITLE
Task 3starting

### PR DESCRIPTION
**Pull request by paulchoi12**

### CODE REVIEW

**Review by yalicee — CHANGES_REQUESTED** [review]
I don't see a 404 error handler for incorrect paths or 500 internal server in your app.js file. also remove all console.logs before pushing

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+})
+
+describe("app!!", ()=>{
+    test('should return all the topics', ()=>{
+        return query(app)
+        .get('/api/topics')
+        .expect(200)
+        .then((response)=>{
+            const {topic} = response.body
+            return response.body
```
Comment (aloush-dev) [inline __tests__/app.test.js]: We don't have any expect statements here so we're not actually testing the response which we are getting back.
lets compare the topic which we have got back and expect it to look how we are expecting.

File: controllers/topic.controllers.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+const { getAllTopics } = require("../models/topic.model")
+
+exports.controller = (req, res) =>{
```
Comment (aloush-dev) [inline controllers/topic.controllers.js]: I'd rename this as we'll be creating more controllers the more tasks we do.

File: controllers/topic.controllers.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    
+    const query = req.query
+    return getAllTopics(query)
```
Comment (aloush-dev) [inline controllers/topic.controllers.js]: Are we using a query here?

File: models/topic.model.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+// the database has to be from the connection
+const db = require("../db/connection")
+
+exports.getAllTopics = (queryFromURL) => {
```
Comment (aloush-dev) [inline models/topic.model.js]: we pass this query through but its never used, do we need it?

**Review by aloush-dev — CHANGES_REQUESTED** [review]
Nice job implementing feedback.
If you're going to extract your error handlers I'd put them in a seperate error handling file.
They wouldn't technically be controllers under MVC.
Please change your test so that we are testing the actual data that comes back, then it will be good to go.

File: __tests__/app.test.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+              {
+                description: 'Not dogs',
+                slug: 'cats'
+              },
+              {
+                description: 'what books are made of',
+                slug: 'paper'
+              }]
+            
+              console.log(response.body)
```
Comment (itskatherine) [inline __tests__/app.test.js:33]: console.log begone 🗑️

**Review by itskatherine — COMMENTED** [review]
Hi Jiwoo, get rid of the console.log then you're good to merge! 🙌

**Pull request merged**